_source/_posts/2022-04-05-state-of-ffi-java.md

```diff
@@ -0,0 +1,280 @@
+---
+layout: blog_post
+title: "Does Java finally have a secure alternative to JNI?"
+author: deepu-sasidharan
+by: advocate
+communities: [java]
+description: "Let's look at the state of Foreign Function Interface (FFI) in Java."
+tags: [java, security]
+tweets:
+  - "Does #Java finally have a secure alternative to JNI? Let's find out and take a deep look into FFI in Java."
+  - "What is the secure alternative to JNI in Java 18? Let's find out #java #security"
+image: blog/state-of-ffi-java/cover.png
+type: awareness
+---
+
+Java 18 was released last month (March 2022), and with it comes the second incubator of the Foreign Function & Memory API so let us look at the state of Foreign Function Interface (FFI) in Java.
+
+{% include toc.md %}
+
+If you would instead follow along by watching a video, check out the video of the talk I made on the same topic, at FOSDEM'22, below from [the OktaDev YouTube channel](https://youtu.be/lW69_AtAXzE).
+
+{% youtube lW69_AtAXzE %}
+
+## What is a Foreign Function Interface?
+
+A foreign function interface is the ability to call functions or routines written in one programming language from another programming language. This is generally used to access native functions or programs on the host OS written in low-level languages like C. Most languages provide this feature out of the box in some forms. The term originated from common LISP, but it's known by different names in different languages. Most languages use the C/C++ calling conventions for FFI and natively support calling C/C++ functions.
+
+## Why is Foreign Function Interface needed?
+
+Most of the use cases for FFI were around interacting with legacy apps and accessing host OS features or native libraries. But the recent surge in machine learning and advanced arithmetics made FFI even more necessary. These days we use foreign functions for an array of use cases, some of which are:
+
+- Interact with legacy apps
```

Comment:
```suggestion
- Interacting with legacy apps
```
changing these verbs to gerund (-ing form) so that the bullets 'complete' the sentence above